openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -239,37 +241,76 @@ def tearDown(self):
     )
     @ddt.unpack
-    def test_home_page(self, user_type, expected_message):
+    def test_home_page(
+            self, enable_unenrolled_access, course_visibility, user_type,
+            expected_enroll_message, expected_course_outline,
+    ):
         self.create_user_for_course(self.course, user_type)
 
         # Render the course home page
-        url = course_home_url(self.course)
-        response = self.client.get(url)
+        with mock.patch('xmodule.course_module.CourseDescriptor.course_visibility', course_visibility):
+            # Test access with anonymous flag and course visibility
+            with override_waffle_flag(COURSE_ENABLE_UNENROLLED_ACCESS_FLAG, enable_unenrolled_access):
+                url = course_home_url(self.course)
+                response = self.client.get(url)
 
         # Verify that the course tools and dates are always shown
         self.assertContains(response, TEST_COURSE_TOOLS)
         self.assertContains(response, TEST_COURSE_TODAY)
 
-        # Verify that the outline, start button, course sock, and welcome message
-        # are only shown to enrolled users.
+        is_anonymous = user_type is CourseUserType.ANONYMOUS
         is_enrolled = user_type is CourseUserType.ENROLLED
-        is_unenrolled_staff = user_type is CourseUserType.UNENROLLED_STAFF
-        expected_count = 1 if (is_enrolled or is_unenrolled_staff) else 0
-        self.assertContains(response, TEST_CHAPTER_NAME, count=expected_count)
-        self.assertContains(response, 'Start Course', count=expected_count)
+        is_enrolled_or_staff = is_enrolled or user_type in (
+            CourseUserType.UNENROLLED_STAFF, CourseUserType.GLOBAL_STAFF
+        )
+
         self.assertContains(response, 'Learn About Verified Certificate', count=(1 if is_enrolled else 0))
-        self.assertContains(response, TEST_WELCOME_MESSAGE, count=expected_count)
+
+        # Verify that start button, course sock, and welcome message
+        # are only shown to enrolled users or staff.
+        self.assertContains(response, 'Start Course', count=(1 if is_enrolled_or_staff else 0))
+        self.assertContains(response, TEST_WELCOME_MESSAGE, count=(1 if is_enrolled_or_staff else 0))
+
+        # Verify the outline is shown to enrolled users, unenrolled_staff and anonymous users if allowed
+        self.assertContains(response, TEST_CHAPTER_NAME, count=(1 if expected_course_outline else 0))
 
         # Verify that the expected message is shown to the user
-        self.assertContains(response, '<div class="user-messages"', count=1 if expected_message else 0)
-        if expected_message:
-            self.assertContains(response, expected_message)
+        self.assertContains(
+            response, 'To see course content', count=(1 if is_anonymous else 0)
+        )
+        self.assertContains(response, '<div class="user-messages"', count=(1 if expected_enroll_message else 0))
```

Comment:
Fixed conflict on this line because of https://github.com/edx/edx-platform/commit/67e9ae55817dee5d8b04f132515cd2bc55daaf42#diff-a9b73fc05b659b3ec3afab5a6f47b628R269